commit f6e57fc0793f0196530f74484a16c70c0dcd8aa3
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Wed Nov 2 12:53:45 2022 +0100

    HPC2N specific patch
    Add "account" and GPU specification arguments to the GUI,
    and gres arguments to the submit file.

diff --git a/src/gui_jobwindow.cpp b/src/gui_jobwindow.cpp
index 2a8f374f..c73004d1 100644
--- a/src/gui_jobwindow.cpp
+++ b/src/gui_jobwindow.cpp
@@ -157,7 +157,10 @@ void JobWindow::setupRunTab()
 	place("qsub");
 	place("queuename");
 
+        place("account");
 	place("timelimit");
+        place("gputype");
+        place("ngpu");
 	place("extrasbatch");
 
 	char * extra_count_text = getenv ("RELION_QSUB_EXTRA_COUNT");
diff --git a/src/pipeline_jobs.cpp b/src/pipeline_jobs.cpp
index dee33dba..af0ac9de 100644
--- a/src/pipeline_jobs.cpp
+++ b/src/pipeline_jobs.cpp
@@ -586,11 +586,24 @@ bool RelionJob::saveJobSubmissionScript(std::string newfilename, std::string out
                 } else {
                     replacing["XXXqueueXXX"] = "";
                 }
+                if (!joboptions["account"].getString().empty()) {
+                    replacing["XXXaccountXXX"] = "-A " + joboptions["account"].getString();
+                } else {
+                    replacing["XXXaccountXXX"] = "";
+                }
                 if (!joboptions["timelimit"].getString().empty()) {
                     replacing["XXXtimelimitXXX"] = "-t " + joboptions["timelimit"].getString();
                 } else {
                     replacing["XXXtimelimitXXX"] = "";
                 }
+                if (!joboptions["gputype"].getString().empty()) {
+                    replacing["XXXgpuspecXXX"] = "--gres=gpu:" + joboptions["gputype"].getString();
+                    if (!joboptions["ngpu"].getString().empty()) {
+                        replacing["XXXgpuspecXXX"] += ":" + joboptions["ngpu"].getString();
+                    }
+                } else {
+                    replacing["XXXgpuspecXXX"] = "";
+                }
                 replacing["XXXextrasbatchXXX"] = joboptions["extrasbatch"].getString();
 		char *extra_count_text = getenv("RELION_QSUB_EXTRA_COUNT");
 		const char extra_count_val = (extra_count_text ? atoi(extra_count_text) : 2);
@@ -954,9 +967,22 @@ the job will be executed locally. Note that only MPI jobs may be sent to a queue
 Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
 (or create your own script following the RELION Wiki) if you have trouble submitting jobs. The default command can be set through the environment variable RELION_QSUB_COMMAND.");
 
+        joboptions["account"] = JobOption("Account:", std::string(""), "SNIC Account (project id).\n\n\
+Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
+(or create your own script following the RELION Wiki) if you have trouble submitting jobs.");
+
         joboptions["timelimit"] = JobOption("Timelimit - format days-hh:mm:ss:", std::string(""), "Timelimit requested for the job.\n\n\
 Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
 (or create your own script following the RELION Wiki) if you have trouble submitting jobs.");
+
+        joboptions["gputype"] = JobOption("GPU type - k80 or v100:", std::string(""), "Which type of GPU to use for GPU jobs.\n\n\
+Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
+(or create your own script following the RELION Wiki) if you have trouble submitting jobs.");
+
+        joboptions["ngpu"] = JobOption("Number of GPU cards to use:", std::string(""), "How many GPU cards of the specified type to use.\n\n\
+Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
+(or create your own script following the RELION Wiki) if you have trouble submitting jobs.");
+
         joboptions["extrasbatch"] = JobOption("Other sbatch arguments:", std::string(""), "Other arguments to sbatch you want to specify.\n\n\
 Note that the person who installed RELION should have made a custom script for your cluster/queue setup. Check this is the case \
 (or create your own script following the RELION Wiki) if you have trouble submitting jobs.");
diff --git a/src/gui_entries.h b/src/gui_entries.h
index 87741dad..b732d55b 100644
--- a/src/gui_entries.h
+++ b/src/gui_entries.h
@@ -85,10 +85,10 @@ extern bool create_scheduler_gui;
 #define MENUHEIGHT 30
 #define TABHEIGHT 25
 #define GUIWIDTH 800
-#define GUIHEIGHT_OLD 420+GUIEXTRA
-#define GUIHEIGHT_EXT_START 370+GUIEXTRA
+#define GUIHEIGHT_OLD 440+GUIEXTRA
+#define GUIHEIGHT_EXT_START 390+GUIEXTRA
 #define GUIHEIGHT_EXT_START2 (GUIHEIGHT_EXT_START+MENUHEIGHT+10)
-#define GUIHEIGHT_EXT 800+GUIEXTRA
+#define GUIHEIGHT_EXT 820+GUIEXTRA
 #define XCOL0 200
 #define WCOL0 200
 #define XCOL1 ( (XCOL0) + 10  )
